### PR TITLE
fix: update elevation ascii workflow TDE-1050

### PR DESCRIPTION
**Pull request by MDavidson17**
#### Motivation

update ascii-standardising workflow so that it is ready to publish elevation data

#### Modification

update the workflow to use:
- argo-tasks v3
- topo-imagery v4 **nb: this version of topo-imagery makes the translate-ascii step quite slow I will investigate this however this PR will work**
- publish-odr 

#### Checklist

_If not applicable, provide explanation of why._

- [x] Tests updated - tested manually up to create-manifest 
- [x] Docs updated
- [x] Issue linked in Title

### CODE REVIEW

File: workflows/raster/README.md [hunk 1]
```diff
@@ -424,6 +425,89 @@ aligned-level: '6'
 create-pull-request: 'true'
 ```
 
+# ascii-standardise-publish
+
+**For command line use only**
+
+This is a specific Workflow intended for bulk elevation transfers which are currently stored in linz-elevation-staging as .asc files and do not require Visual QA before publication.
```
Comment (paulfouquet) [inline workflows/raster/README.md:432]:
```suggestion
This is a specific Workflow intended for bulk elevation transfers which are currently stored in `s3://linz-elevation-staging/` as `.asc` files and do not require Visual QA before publication.
```

File: workflows/raster/README.md [hunk 2]
```diff
@@ -424,6 +425,89 @@ aligned-level: '6'
...
+
+## Workflow Input Parameters
+
+### ascii-standardise-publish Mandatory Parameters - must be specified on the command line
+
+| Parameter      | Type | Default | Description                                                                                                                                                     |
+| -------------- | ---- | ------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| ticket         | str  |         | Ticket ID e.g. 'AIP-55'                                                                                                                                         |
+| region         | enum |         | Region of the dataset :warning: The name has to be exactly one in the region enum in `standardising`.                                                           |
+| source         | str  |         | the uri (path) to the input tiffs e.g. s3://linz-imagery-upload/test/sample                                                                                     |
```
Comment (paulfouquet) [inline workflows/raster/README.md:444]:
```suggestion
| source         | str  |         | The uri (path) to the input tiffs e.g. s3://linz-imagery-upload/test/sample                                                                                     |
```

File: workflows/raster/README.md [hunk 3]
```diff
@@ -424,6 +425,89 @@ aligned-level: '6'
...
+| target         | str  |         | the uri (path) to the published tiffs in the format s3://linz-imagery-target-example/region/city-or-sub-region_year_resolution/product/crs/                     |
```
Comment (paulfouquet) [inline workflows/raster/README.md:445]:
```suggestion
| target         | str  |         | The uri (path) to the published tiffs in the format s3://linz-imagery-target-example/region/city-or-sub-region_year_resolution/product/crs/                     |
```

File: workflows/raster/README.md [hunk 4]
```diff
@@ -424,6 +425,89 @@ aligned-level: '6'
...
+| category       | enum |         | Dataset type for collection metadata, also used to Build Dataset title & description                                                                            |
+| gsd            | str  |         | dataset GSD required for dataset title                                                                                                                          |
```
Comment (paulfouquet) [inline workflows/raster/README.md:447]:
```suggestion
| gsd            | str  |         | Dataset GSD required for dataset title                                                                                                                          |
```

File: workflows/raster/README.md [hunk 5]
```diff
@@ -424,6 +425,89 @@ aligned-level: '6'
...
+| cutline                | str   |              | (Optional) location of a cutline file to cut the imagery to `.fgb` or `.geojson` (do not include if no cutline)                                                                                                                                                                                                      |
+| collection_id          | str   |              | (Optional) Provide a Collection ID if re-processing an existing published survery, otherwise a ULID will be generated for the collection.json ID field.                                                                                                                                                              |
+| compression            | str   | webp         | Standardised file format. Must be `webp` or `lzw`                                                                                                                                                                                                                                                                    |
```
Comment (paulfouquet) [inline workflows/raster/README.md:461]: as discussed

File: workflows/raster/ascii-standardise-publish.yaml [hunk 6]
```diff
@@ -127,16 +127,18 @@ spec:
...
             arguments:
               parameters:
                 - name: source
-                  value: '{{tasks.get-location.outputs.parameters.location}}flat/'
+                  value: '{{tasks.standardise.outputs.parameters.target}}flat/'
                 - name: include
-                  value: '\.tiff?$|\.json$'
+                  value: '\.tiff?$|\.json$|/capture-area\.geojson$'
+                - name: target_bucket_name
+                  value: '{{workflow.parameters.target_bucket_name}}'
```
Comment (paulfouquet) [inline workflows/raster/ascii-standardise-publish.yaml:141]: Where is this parameter declared?

Comment (MDavidson17) [reply workflows/raster/ascii-standardise-publish.yaml:143]: hmm I've added it I'm not sure how it worked without it but it did :woman_shrugging:

Comment (paulfouquet) [reply workflows/raster/ascii-standardise-publish.yaml:143]: Looks like Argo Workflows is not controlling the parameters you pass to a workflow?

**Review by paulfouquet — APPROVED** [review]

**Review by amfage — APPROVED** [review]

**Review by l0b0 — APPROVED** [review]

**Pull request merged**